internal/release/promote/promotion.go

```diff
@@ -19,11 +17,34 @@ type Promotion struct {
 	PromptProvider      PromptProvider
 	GitProvider         GitProvider
 	PullRequestProvider pr.PullRequestProvider
-	YamlWriter          YamlWriter
+	YamlWriter          yml.Writer
 	CommitTemplate      string
 	PullRequestTemplate string
 	InfoProvider        info.Provider
 	LinksProvider       links.Provider
+
+	// Prompt is the prompt to use for user interaction.
+	promptProvider PromptProvider
+
+	// Committer allows committing and pushing changes to git.
+	gitProvider GitProvider
+
+	// PullRequestProvider is the provider of pull requests.
+	pullRequestProvider pr.PullRequestProvider
+
+	commitTemplate      string
+	pullRequestTemplate string
```

Comment:
Pourquoi tout ça en double? Erreur de merge/résolution de conflit?

Reply:
seems like it! will remove